frontend/pages/beta/model/[modelId]/release/new.tsx

```diff
@@ -19,71 +19,69 @@ export default function NewRelease() {
 
   const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
     event.preventDefault()
-    if (model) {
-      setErrorMessage('')
-      setLoading(true)
-      if (!model.card.version) {
-        setLoading(false)
-        return setErrorMessage('Please make sure your model has a schema set before drafting a release.')
-      }
-      if (isValidSemver(semver)) {
-        const fileIds: string[] = []
-        for (const artefact of artefacts) {
-          const artefactMetadata = artefactsMetadata.find((metadata) => metadata.fileName === artefact.name)
-          const postArtefactResponse = await postFile(
-            artefact,
-            model.id,
-            artefact.name,
-            artefact.type,
-            artefactMetadata?.metadata,
-          )
-          if (postArtefactResponse.ok) {
-            const res = await postArtefactResponse.json()
-            fileIds.push(res.file._id)
-          } else {
-            setLoading(false)
-            return setErrorMessage(await getErrorMessage(postArtefactResponse))
-          }
-        }
 
-        setLoading(false)
+    if (!model) {
+      return
+    }
 
-        const release: CreateReleaseParams = {
-          modelId: model.id,
-          semver,
-          modelCardVersion: model.card.version,
-          notes: releaseNotes,
-          minor: isMinorRelease,
-          fileIds: fileIds,
-          images: imageList,
-        }
+    if (!model.card.version) {
+      return setErrorMessage('Please make sure your model has a schema set before drafting a release.')
+    }
+
+    if (!isValidSemver(semver)) {
+      return setErrorMessage('Please set a valid semver value before drafing a release.')
+    }
 
-        const response = await postRelease(release)
+    setErrorMessage('')
+    setLoading(true)
 
-        if (!response.ok) {
-          const error = await getErrorMessage(response)
-          setLoading(false)
-          return setErrorMessage(error)
-        }
+    const fileIds: string[] = []
+    for (const file of files) {
+      const fileMetadata = filesMetadata.find((metadata) => metadata.fileName === file.name)
+      const postFileResponse = await postFile(file, model.id, file.name, file.type, fileMetadata?.metadata)
 
+      if (!postFileResponse.ok) {
         setLoading(false)
-        mutateReleases()
-        router.push(`/beta/model/${modelId}?tab=releases`)
+        return setErrorMessage(await getErrorMessage(postFileResponse))
       }
+
+      const res = await postFileResponse.json()
+      fileIds.push(res.file._id)
     }
+
+    const release: CreateReleaseParams = {
+      modelId: model.id,
+      semver,
+      modelCardVersion: model.card.version,
+      notes: releaseNotes,
+      minor: isMinorRelease,
+      fileIds: fileIds,
+      images: imageList,
+    }
+
+    const response = await postRelease(release)
+
+    if (!response.ok) {
+      setLoading(false)
+      return setErrorMessage(await getErrorMessage(response))
```

Comment:
The way we were getting the error message and setting the loading state previously is preferred here as we were calling `setLoading(false)` once all of the asynchronous logic had finished